presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1567,6 +1575,46 @@ private Scope computeAndAssignOrderByScope(OrderBy node, Scope sourceScope, Scop
             return orderByScope;
         }
 
+        private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope outputScope, List<FunctionCall> aggregations, List<List<Expression>> groupByExpressions)
+        {
+            // This scope is only used for planning. When aggregation is present then
+            // only output fields, groups and aggregation expressions should be visible from ORDER BY expression
+            ImmutableList.Builder<Expression> orderByAggregationExpressionsBuilder = ImmutableList.builder();
+            if (groupByExpressions.size() == 1) {
+                orderByAggregationExpressionsBuilder.addAll(getOnlyElement(groupByExpressions));
+            }
+            orderByAggregationExpressionsBuilder.addAll(aggregations);
+
+            // Don't add aggregate expression that contains references to output column because the names would clash in TranslationMap during planning.
+            List<Expression> orderByAggregationExpressions = orderByAggregationExpressionsBuilder.build().stream()
+                    .filter(expression -> !hasEqualOrderByExpressionWithOutputColumnReference(expression, node, outputScope))
+                    .collect(toImmutableList());
+
+            // generate placeholder fields
+            ImmutableList.Builder<Field> orderByAggregationSourceFields = ImmutableList.builder();
```

Comment:
Generally, avoid non-functional style for purely functional transformations. I'd do this, instead.

```java
// generate placeholder fields
List<Field> sourceFields = orderByAggregationExpressions.stream()
        .map(analysis::getType)
        .map(type -> Field.newUnqualified(Optional.empty(), type))
        .collect(toImmutableList());
```